aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -204,11 +203,84 @@ func resourceAwsApiGatewayRestApiCreate(d *schema.ResourceData, meta interface{}
 			input.Parameters = stringMapToPointers(v.(map[string]interface{}))
 		}
 
-		_, err := conn.PutRestApi(input)
+		output, err := conn.PutRestApi(input)
 
 		if err != nil {
 			return fmt.Errorf("error creating API Gateway specification: %s", err)
 		}
+
+		// Using PutRestApi with mode overwrite will remove any configuration
+		// that was done with CreateRestApi. Reconcile these changes by having
+		// any Terraform configured values overwrite imported configuration.
+
+		updateInput := &apigateway.UpdateRestApiInput{
+			RestApiId:       aws.String(d.Id()),
+			PatchOperations: []*apigateway.PatchOperation{},
+		}
+
+		if v, ok := d.GetOk("api_key_source"); ok && v.(string) != aws.StringValue(output.ApiKeySource) {
+			updateInput.PatchOperations = append(updateInput.PatchOperations, &apigateway.PatchOperation{
+				Op:    aws.String(apigateway.OpReplace),
+				Path:  aws.String("/apiKeySource"),
+				Value: aws.String(v.(string)),
+			})
+		}
+
+		if v, ok := d.GetOk("binary_media_types"); ok && len(v.([]interface{})) > 0 {
```

Comment:
Perhaps efforts are underway already, but it would be handy if `GetOk()` were modified for lists/sets or a new method were added (maybe `GetOkSest()`? `GetOkLet()`? `GetOkCollection()`? I'm guessing there are many instances in the code base we don't have the `len(v.([]interface{})) > 0` check.